### PR TITLE
use screen width to calculate preview size

### PR DESCRIPTION
**Pull request by DeepDiver1975**
@jancborchardt @georgehrke @PVince81 @karlitschek

### CODE REVIEW

**Comment by georgehrke** [comment]
code looks good, will test asap

File: js/slideshow.js [hunk 1]
```diff
@@ -344,12 +344,14 @@ $(document).ready(function () {
 			for (var i = 0; i < files.length; i++) {
 				var file = files[i];
 				if (file.mimetype && file.mimetype.indexOf('image') >= 0) {
-					var imageUrl = OC.generateUrl('/core/preview.png?file={file}&x=1000&a=true', {
+					var imageUrl = OC.generateUrl('/core/preview.png?file={file}&x={x}&a=true', {
+						x: $(document).width() * window.devicePixelRatio,
```
Comment (PVince81) [inline js/slideshow.js]: Please don't call $(document).width() in a loop but cache it beforehand as it can be expensive.

**Comment by icewind1991** [comment]
Please add `scalingup=0` to prevent upscaling images and wasting even more bandwidth

**Comment by DeepDiver1975** [comment]
@icewind1991 added

@icewind1991 @georgehrke @PVince81 please review and test - THX

**Comment by icewind1991** [comment]
tested, code looks good :+1:

**Comment by PVince81** [comment]
Works :+1: 
Happy to see that my panoramic pictures (a few MB big) are quick to display now :smile:

**Pull request merged**